server/internal/middleware/hooks/gen_cros_hook.go

```diff
@@ -0,0 +1,67 @@
+// Copyright © 2023 OpenIM open source community. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package hooks
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+
+	urltrie "github.com/OpenIMSDK/OpenKF/server/internal/middleware/hooks/url_trie"
+)
+
+var _ urltrie.Hook = (*CROS)(nil)
+
+func init() {
+	urltrie.RegisterHook(&CROS{})
+	fmt.Println("RegisterHook", "Register Hook[CROS] success...")
+}
+
+// CROS implement urltrie.Hook.
+type CROS struct {
+	urltrie.Hook
+}
+
+// Patterns EDIT THIS TO YOUR OWN HOOK PATTERN.
+func (h *CROS) Patterns() []string {
+	return []string{
+		"/*",
+	}
+}
+
+// Priority EDIT THIS TO YOUR OWN HOOK PRIORITY.
+func (h *CROS) Priority() int64 {
+	return 0
+}
+
+// BeforeRun EDIT THIS TO YOUR OWN HOOK BEFORE RUN, DO NOT NEED USE Next() FUNCTION.
+func (h *CROS) BeforeRun(c *gin.Context) {
+	// c.Writer.Header().Set("Access-Control-Allow-Origin", "*")
+	c.Writer.Header().Set("Access-Control-Allow-Origin", c.GetHeader("origin"))
```

Comment:
```suggestion
	c.Writer.Header().Set("Access-Control-Allow-Origin", c.GetHeader("origin"))
```
Whether to hardcode?

Reply:
These hard code are from http request headers, and they may not be used outside.